### ISSUE TITLE
scikit-build ignored when installing sdist

### ISSUE THREAD

**Issue by Jegp** [scikit-build/scikit-build#778]
When creating an sdist (`build -s .`), and subsequently installing the generated tar file, pip does not hook into the scikit-build `setup` function. Instead, pip simply copies the python files and ignores scikit-build.

My directory structure is simple:
```
aestream
  \-- src
  \-- CMakeLists.txt
  \-- setup.py
```

And my `setup.py` looks like this:
```python
from skbuild import setup  # Use scikit-build
cmake_args = [...]
setup(
    name="aestream",
    packages=["aestream"],
    install_requires=install_requires,
    cmake_args=cmake_args
)
```

Since the `cmake_source_dir` picks up the present directory, shouldn't it pick up the `CMakeLists.txt` in the root of the hierarchy?

For anyone interested, the repo is here: https://github.com/norse/aestream/

**Comment by henryiii** [comment]
Could you check the contents of the SDist (and/or wheel)?

```bash
# Show SDist contents
tar -tvf dist/*.tar.gz
# Show wheel contents
unzip -l dist/*.whl
```

**Comment by henryiii** [comment]
Wow, you require torch in the isolated build environment. Though it didn't take that long to run, so guess it's not that bad, I expect it's a large tmp environment, though.

Your MANIFEST.in doesn't include CMakeLists.txt.

**Comment by Jegp** [comment]
Thank you so much for the swift reply! You're completely right, the `CMakeLists.txt` in the *root* directory was not included. It was added from the `src` directory, though. Adding a `include CMakeLists.txt` into `MANIFEST.in` solved the problem.

> Wow, you require torch in the isolated build environment. Though it didn't take that long to run, so guess it's not that bad, I expect it's a large tmp environment, though.

Yep. I need that because I have to link against PyTorch and I wanted to avoid escaping the sandbux. In general, it's been a painful process to get all the moving bits to align... Do you know if efforts are underway for simpler package management for more complex libraries (like Torch)? I am also eager to hear if there are better ways to go about this.

That said, I'm convinced my setup would have been more painful without scikit-build. Thank you for your work :-)

**Comment by henryiii** [comment]
> Do you know if efforts are underway for simpler package management for more complex libraries (like Torch)?

My focus currently is on scikit-build-core. Once that's ready, PyTorch is one of the libraries that will be working on adapting scikit-build-core, so hopefully this will improve. Though if you need a link against a library, I'm not sure if there's another way - though I'd generally think you want to dynamically load at runtime - the version of PyTorch a user gets won't always be the one you pull when building.

**Comment by Jegp** [comment]
> Though if you need a link against a library, I'm not sure if there's another way - though I'd generally think you want to dynamically load at runtime - the version of PyTorch a user gets won't always be the one you pull when building.

True, that would be the ideal. Unfortunately, the version of PyTorch you want to link against, changes depending on your hardware (ignoring ABI for now). Do you know if there are plans to address that incompatibility? My naïve understanding tells me that the only solution to that dilemma is to produce wheels for each possible permutation of PyTorch version and hardware (CPU/GPU) setup.
